### DISCUSSION TITLE
Dịch garbage collector, scope, persistent

### DISCUSSION THREAD

**Discussion by minhduc0711** [mlbvn/handson-ml2-vn#130] [Terminology] [answered]
Một vài từ liên quan đến lập trình:
* garbage collector: bộ thu gom rác
* scope: phạm vi
* persistent/persistance: từ này hơi khoai :sweat: 
  Context:
  >   If you need to call `gradient()` more than once, you must make the tape **persistent** and delete it each time you are done with it to free resources.
  >   If the tape goes out of scope, for example when the function that used it returns, Python’s garbage collector will delete it for you.
  >   ```python
  >   with tf.GradientTape(persistent=True) as tape:
  >       z = f(w1, w2)
  >   
  >   dz_dw1 = tape.gradient(z, w1) # => tensor 36.0
  >   dz_dw2 = tape.gradient(z, w2) # => tensor 10.0, works fine now!
  >   del tape
  >   ```

cc @mlbvn/handson-ml

**Comment by lkhphuc** [comment]
`make gradient tape persistent` mình có thể dịch thành `duy trì cuộn băng gradient...`.

**Reply by minhduc0711** [reply]
> Keras will take care of variable persistence seamlessly; no action is required.

Có mấy chỗ như này nữa, ý nói `tf.Variable` được lưu lại cùng mô hình

**Reply by rootonchair** [reply]
từ "persistent" này có vẻ như không có nghĩa cứng nhỉ?

**Reply by minhduc0711** [reply]
Mình hiểu nghĩa cứng là "được lưu lại vào đâu đó nên không bị mất"

**Reply by minhduc0711** [reply]
có thể tạm dịch là "sao lưu" được không nhỉ?

**Reply by cuongvng** [reply]
Chắc từ này tùy context để dịch cho xuôi thôi a, ko nên fix cứng. Giữ đc nghĩa là ổn.

**Reply by minhduc0711** [reply]
Ừm vậy mình cứ để "persistence" trong glossary là "việc sao lưu / duy trì" nhé, tùy context để mọi người chọn (có thể bổ sung thêm nếu cần).

**Comment by cuongvng** [comment]
Garbage collector thành "bộ dọn rác" cho gọn ổn không nhỉ?

**Reply by minhduc0711** [reply]
Mình thấy "thu gom" có vẻ được dùng phổ biến hơn

**Reply by cuongvng** [reply]
Vậy cứ chốt là thu gom đi ạ.

**Comment by minhduc0711** [comment, marked as answer]
Chốt nhé mn
* garbage collector: bộ thu gom rác
* scope: phạm vi
* persistence: sao lưu / duy trì
